### PR TITLE
Remove build warnings

### PR DESCRIPTION
**Pull request by mbanth**

### CODE REVIEW

**Review by lucianomartin — APPROVED** [review]
Changes are addressing potential warnings, but I can't comment if they are technically correct as I am not very familiar with the code.

Considering the scope of the changes, I can't assess either if adding some info in the  CHANGELOG is necessary.

File: lib_xud/src/core/XUD_DeviceAttach.xc [hunk 1]
```diff
@@ -149,5 +149,7 @@ int XUD_DeviceAttachHS(XUD_PwrConfig pwrConfig)
                 break;
         }
     }
+
+    return -1;
```
Comment (xross) [inline lib_xud/src/core/XUD_DeviceAttach.xc:153]: Not sure on the value but should be unreachable

File: lib_xud/src/core/XUD_HAL.xc [hunk 2]
```diff
@@ -43,6 +43,8 @@ unsigned XtlSelFromMhz(unsigned m)
             while(1);
             break;
     }
+
+    return 0b000;
```
Comment (xross) [inline lib_xud/src/core/XUD_HAL.xc:47]: Fine (though I think putting the return in the default will make the compiler happy)

File: lib_xud/src/core/XUD_HAL.xc [hunk 3]
```diff
@@ -340,6 +342,7 @@ XUD_LineState_t XUD_HAL_GetLineState(/*XUD_HAL_t &xudHal*/)
     if(se0)
         return XUD_LINESTATE_SE0;
 
+    return XUD_LINESTATE_SE1;
```
Comment (xross) [inline lib_xud/src/core/XUD_HAL.xc:345]: I believe this should be fine.

File: lib_xud/src/core/XUD_HAL.xc [hunk 4]
```diff
@@ -375,6 +378,7 @@ unsigned XUD_HAL_WaitForLineStateChange(XUD_LineState_t &currentLs, unsigned tim
     return 0;
 #else
     #warning TODO for XS2A
+    return 1;
```
Comment (xross) [inline lib_xud/src/core/XUD_HAL.xc:381]: Fine for now.

File: lib_xud/src/core/XUD_Signalling.xc [hunk 5]
```diff
@@ -40,9 +40,6 @@ extern unsigned g_curSpeed;
 
 int XUD_Init()
```
Comment (xross) [inline lib_xud/src/core/XUD_Signalling.xc:41]: Good

**Review by xross — APPROVED** [review]

**Pull request merged**